Curric/unit modal CUR-267

k-huggs opened the pull request:
## Curriculum Modal (MVP)

- Adds Sidebar component and test file
- Lesson Metadata component and test file
- Adds Unit Modal component and test file
- Adds Unit Info button to Unit Tab component

## Issue(s)

Fixes #CUR-267

## How to test

1. Go to https://deploy-preview-1953--oak-web-application.netlify.thenational.academy/teachers/curriculum/english-secondary-aqa/units
2. Click on unit info buttons to open modal
3. You should see unit info for that specific modal along with the LessonMetadata component
4. Click on a unit with optional units 
5. You should se unit optionality card and coming soon buttons to access those units

## Screenshots

How it used to look (delete if n/a):
<img width="934" alt="Screenshot 2023-09-29 at 18 23 15" src="https://github.com/oaknational/Oak-Web-Application/assets/91190841/fd409f5c-0642-4af8-9c9c-6fab681487c9">

How it should now look:
<img width="798" alt="Screenshot 2023-10-02 at 11 56 17" src="https://github.com/oaknational/Oak-Web-Application/assets/91190841/62249cc7-0fc2-4c80-b0e2-069ad9dc547d">
<img width="756" alt="Screenshot 2023-10-02 at 11 56 05" src="https://github.com/oaknational/Oak-Web-Application/assets/91190841/051f8500-6005-4691-a8bf-5d08edac7dac">
<img width="742" alt="Screenshot 2023-10-02 at 11 55 46" src="https://github.com/oaknational/Oak-Web-Application/assets/91190841/13f4c807-5748-431a-b2c2-1fbf3361fb15">



## Checklist

- [x] Added or updated tests where appropriate
- [x] Manually tested across browsers / devices
- [x] Considered impact on accessibility
- [x] Design sign-off
- [ ] Approved by product owner

HelenHarlow reviewed (changes requested):
Exciting to see this coming together! A few UI tweaks from me:

- Can we add some spacing between the coming soon tag and the button (in designs this is 8px)
- Is it possible to remove the hover state on the disabled buttons? Just to help show they aren’t clickable
- Could we remove the svg border around the pink box when there are unit options? We need it on the unit cards but the pink box can match the unit visualiser (straight container with 4px corner radius)
- Last one - sorry I know I asked for the tag & button to be horizontally aligned, but because the tag wraps onto 2 lines it looks kind of funny 😅 Could we change it to vertical alignment with the tag and button aligned? (screenshot attached)

<img width="660" alt="Screenshot 2023-10-02 at 12 40 52" src="https://github.com/oaknational/Oak-Web-Application/assets/144228131/795888c2-bd34-407a-8542-355440b7a239">

rightsaidjames reviewed (changes requested):
In addition to Helen's changes above (the first is a straightforward bug, the other two might need some input from @HelenHarlow):

* When a unit is highlighted, the `Unit info` text stays black and is therefore not visible.
* When unit titles for units with options stretch over more than one line, the spacing on the tile is a bit tight/uneven.
* For units without options, the `Unit info` text is obscured by the feedback widget (including on mobile)

<img width="996" alt="Screenshot 2023-10-02 at 12 57 50" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/da603e87-f00e-4109-9275-c0a4aa8785c7">
<img width="996" alt="Screenshot 2023-10-02 at 12 56 54" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/cf08b5ff-376a-4736-9696-69fc260865b1">
<img width="531" alt="Screenshot 2023-10-02 at 13 00 22" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/7e948c45-1d0a-4982-a08a-f3a188a5635a">
<img width="1679" alt="Screenshot 2023-10-02 at 14 14 44" src="https://github.com/oaknational/Oak-Web-Application/assets/3018395/cfec84c3-fde4-4b99-9b76-c60606786640">

sonalisin commented on src/components/LessonMetadata/LessonMetadata.tsx:
```diff
@@ -5,21 +5,26 @@ import Flex from "../Flex";
-      {i + 1 !== metadata.length && (
-        <Typography aria-hidden={true}>•</Typography>
-      )}
+      {i + 1 !== metadata.length && <Typography>•</Typography>}
```
I think this might be preference but `i < metadata.length` might be a tiny bit clearer

k-huggs replied:
changed ✅

sonalisin commented on src/components/Sidebar/Sidebar.test.tsx:
```diff
@@ -0,0 +1,75 @@
+  test("should render the sidebar with children", () => {
+    const { getByTestId } = renderWithTheme(
+      <Sidebar displayModal={true} onClose={jest.fn()}>
+        <div data-testid="sidebar-children" />
```
Not sure what this line is doing - does this also get rendered in the component?

k-huggs replied:
The sidebar should render the children of it, so this test is to check to see if that div is present

sonalisin commented on src/components/UnitModal/UnitModal.tsx:
```diff
@@ -0,0 +1,154 @@
+};
+
+const UnitModal: FC<UnitModalProps> = ({ unitData }) => {
+  if (!unitData) return null;
```
Should this be in the JSX return statement? like a `unitData ? <><> : null` It might be overlooked up at the top of the component

k-huggs replied:
So this is a type guard for the unitOptionsAvailable boolean being constructed, not the rendering of the component (weird I know)  it's being removed on my next PR

sonalisin reviewed (commented):
LGTM! A few tiny comments on clarity

k-huggs commented:
> In addition to Helen's changes above (the first is a straightforward bug, the other two might need some input from @HelenHarlow):
> 
> * When a unit is highlighted, the `Unit info` text stays black and is therefore not visible.
> * When unit titles for units with options stretch over more than one line, the spacing on the tile is a bit tight/uneven.
> * For units without options, the `Unit info` text is obscured by the feedback widget (including on mobile)
> 
> <img alt="Screenshot 2023-10-02 at 12 57 50" width="996" src="https://user-images.githubusercontent.com/3018395/271978609-da603e87-f00e-4109-9275-c0a4aa8785c7.png"> <img alt="Screenshot 2023-10-02 at 12 56 54" width="996" src="https://user-images.githubusercontent.com/3018395/271978663-cf08b5ff-376a-4736-9696-69fc260865b1.png"> <img alt="Screenshot 2023-10-02 at 13 00 22" width="531" src="https://user-images.githubusercontent.com/3018395/271978774-7e948c45-1d0a-4982-a08a-f3a188a5635a.png"> <img alt="Screenshot 2023-10-02 at 14 14 44" width="1679" src="https://user-images.githubusercontent.com/3018395/271979046-cfec84c3-fde4-4b99-9b76-c60606786640.png">


> In addition to Helen's changes above (the first is a straightforward bug, the other two might need some input from @HelenHarlow):
> 
> * When a unit is highlighted, the `Unit info` text stays black and is therefore not visible.
> * When unit titles for units with options stretch over more than one line, the spacing on the tile is a bit tight/uneven.
> * For units without options, the `Unit info` text is obscured by the feedback widget (including on mobile)
> 
> <img alt="Screenshot 2023-10-02 at 12 57 50" width="996" src="https://user-images.githubusercontent.com/3018395/271978609-da603e87-f00e-4109-9275-c0a4aa8785c7.png"> <img alt="Screenshot 2023-10-02 at 12 56 54" width="996" src="https://user-images.githubusercontent.com/3018395/271978663-cf08b5ff-376a-4736-9696-69fc260865b1.png"> <img alt="Screenshot 2023-10-02 at 13 00 22" width="531" src="https://user-images.githubusercontent.com/3018395/271978774-7e948c45-1d0a-4982-a08a-f3a188a5635a.png"> <img alt="Screenshot 2023-10-02 at 14 14 44" width="1679" src="https://user-images.githubusercontent.com/3018395/271979046-cfec84c3-fde4-4b99-9b76-c60606786640.png">

Fixed highlighting
Added spacing between Unit options label and button
moved view lessons button to the bottom left - Can't change Gleap position ✅

k-huggs commented:
> Exciting to see this coming together! A few UI tweaks from me:
> 
> * Can we add some spacing between the coming soon tag and the button (in designs this is 8px)
> * Is it possible to remove the hover state on the disabled buttons? Just to help show they aren’t clickable
> * Could we remove the svg border around the pink box when there are unit options? We need it on the unit cards but the pink box can match the unit visualiser (straight container with 4px corner radius)
> * Last one - sorry I know I asked for the tag & button to be horizontally aligned, but because the tag wraps onto 2 lines it looks kind of funny 😅 Could we change it to vertical alignment with the tag and button aligned? (screenshot attached)
> 
> <img alt="Screenshot 2023-10-02 at 12 40 52" width="660" src="https://user-images.githubusercontent.com/144228131/271959922-795888c2-bd34-407a-8542-355440b7a239.png">

A lot of these tag issues will be resolved in the next PR as it will be removed so won't make the amendments for now 🙂

rightsaidjames reviewed (approved):
Looking good! All of my outstanding issues are resolved, and as far as I can see the only outstanding issues relate to the 'coming soon' tag on the unit options tile, which will be resolved when we introduce lesson titles to the modal.

HelenHarlow reviewed (approved):
Looks good!

oak-machine-user commented:
:tada: This PR is included in version 1.262.3 :tada:

The release is available on [GitHub release](https://github.com/oaknational/Oak-Web-Application/releases/tag/v1.262.3)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
